src/de/dkfz/b080/co/knowledge/metadata/sampleextractorstrategies/SampleFromFilenameExtractor.groovy

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2018 German Cancer Research Center (Deutsches Krebsforschungszentrum, DKFZ).
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/DKFZ-ODCF/COWorkflowsBasePlugin/LICENSE.txt).
+ */
+package de.dkfz.b080.co.knowledge.metadata.sampleextractorstrategies
```

Comment:
Consider making this a COMetadataAccessor subclass.

But maybe this is sufficient anyway. There may also come the situation, where other metadata is to be extracted from other places. One could configure the COMetadataAccessor with multiple strategies for extracting metadata from a specific place (either each datum individually from a different place, or via a builder that adds all matching places to the accessor, e.g. 

`COMetadataAccessor.create(version1FilesystemSampleStrategy, version1FilesystemPairedEndStrategy, ...)`

After that all metadata would be automatically extracted from whatever source(s) are configured. Even cascading sources would be possible, e.g. if available get from filesystem, otherwise from XML, otherwise from database; or whatever.